tools/slice_image/slice_image.py

```diff
@@ -9,24 +9,29 @@
 import skimage.feature
 from scipy.stats import entropy as scipy_entropy
 
-def slice_image(input_file, out_folder, label=None, label_out_folder=None, window_size=64, stride=1, bg_thresh=1, limit_slices=False, n_thresh=5000):
+def slice_image(input_file, out_folder, label=None, label_out_folder=None, window_size=64, 
+                stride=1, bg_thresh=1, limit_slices=False, n_thresh=5000, seed=None):
     #TODO NOT Implemented:process labels
+    # --> label and label_out_folder useless so far
+
+    # primarily for testing purposes:
+    if seed is not None:
+        random.seed(seed)
 
     img_raw = skimage.io.imread(input_file)
     if len(img_raw.shape) == 2:
         img_raw = np.expand_dims(img_raw, 3)
 
-    patches_raw = skimage.util.view_as_windows(img_raw, (window_size, window_size, img_raw.shape[2]), step=stride)
-    patches_raw = patches_raw.reshape([-1, window_size, window_size, img_raw.shape[2]])
-
-    filename = os.path.splitext(os.path.basename(input_file))[0]
-    new_path = out_folder+"/"+filename+"_%d.tiff"
-    
-    #samples for thresholding the amount of slices
-    sample = random.sample(range(patches_raw.shape[0]), n_thresh)
-
-    with warnings.catch_warnings():
+    with warnings.catch_warnings(): # ignore FutureWarning

```

Comment:
is this all because Galaxy fails when this warning is printed?
If so you can use https://docs.galaxyproject.org/en/latest/dev/schema.html#error-detection `detect_errors="agressive"`...